scalafix-cli/src/main/scala/scalafix/internal/v1/MainOps.scala

```diff
@@ -347,6 +348,18 @@ object MainOps {
     } else {
       args.rules.syntacticPatch(doc, args.args.autoSuppressLinterErrors)
     }
+    // close global if evaluated
+    cleanGlobalRessources(args.global)
```

Comment:
should that be in a finally block instead?

Reply:
yes. but where?

Reply:
I don't understand your question - here?

Reply:
```
try {
 val tree = LazyValue.later { () =>
      args.parse(input).get: Tree
    }
    val doc = SyntacticDocument(input, tree, args.diffDisable, args.config)
    val results = if (args.rules.isSemantic) {
      val relpath = file.toRelative(args.sourceroot)
      val sdoc = SemanticDocument.fromPath(
        doc,
        relpath,
        args.classLoader,
        args.symtab,
        () => compileWithGlobal(args, doc)
      )
      val result =
        args.rules.semanticPatch(sdoc, args.args.autoSuppressLinterErrors)
      assertFreshSemanticDB(
        input,
        file,
        result.fixed,
        sdoc.internal.textDocument
      )
      result
    } else {
      args.rules.syntacticPatch(doc, args.args.autoSuppressLinterErrors)
    }
    results
  } finally {
      // close global if evaluated
    shutdownCompiler(args.global)
    }
```

Reply:
something like this ?

Reply:
hard to tell without proper indentation and highlighting but yes :+1:  (the `results` val can be removed)